### ISSUE TITLE
C library 'SDL2' not found while cross compiling

### ISSUE THREAD

**Issue by Erisaren** [Genymobile/scrcpy#804]
Greetings, I have an issue while trying to cross compile for windows.

I used _sudo make -f Makefile.CrossWindows build-win64_ and _sudo ./release.sh_ with no results.
I always get _app/meson.build:45:4: ERROR: C library 'SDL2' not found_ when I try to compile
and _Makefile.CrossWindows:78: recipe for target 'build-win64' failed
make: *** [build-win64] Error 1_.
I downloaded SDL2-2.0.10 and I unzipped it on prebuilt-deps folder because it was failing to find it.

This is the meson log:

> Build started at 2019-09-14T11:35:17.902871
> Main binary: /usr/bin/python3
> Python system: Linux
> The Meson build system
> Version: 0.45.1
> Source dir: /home/erisa/scrcpy
> Build dir: /home/erisa/scrcpy/build-win64
> Build type: cross build
> Project name: scrcpy
> Sanity testing C compiler: cc
> Is cross compiler: False.
> Sanity check compiler command line: cc /home/erisa/scrcpy/build-win64/meson-private/sanitycheckc.c -o /home/erisa/scrcpy/build-win64/meson-private/sanitycheckc.exe
> Sanity check compile stdout:
> 
> -----
> Sanity check compile stderr:
> 
> -----
> Running test binary command: /home/erisa/scrcpy/build-win64/meson-private/sanitycheckc.exe
> Sanity testing C compiler: /usr/bin/x86_64-w64-mingw32-gcc
> Is cross compiler: True.
> Sanity check compiler command line: /usr/bin/x86_64-w64-mingw32-gcc -c /home/erisa/scrcpy/build-win64/meson-private/sanitycheckc.c -o /home/erisa/scrcpy/build-win64/meson-private/sanitycheckc_cross.exe
> Sanity check compile stdout:
> 
> -----
> Sanity check compile stderr:
> 
> -----
> Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
> Cross C compiler: /usr/bin/x86_64-w64-mingw32-gcc (gcc 7.3.0)
> Host machine cpu family: x86
> Host machine cpu: x86_64
> Target machine cpu family: x86
> Target machine cpu: x86_64
> Build machine cpu family: x86_64
> Build machine cpu: x86_64
> 
> app/meson.build:45:4: ERROR: C library 'SDL2' not found
>

**Comment by rom1v** [comment]
> `sudo ./release.sh` with no results.

And without `sudo`? You must not be root to build.

**Comment by Erisaren** [comment]
Thank you for your reply.
I tried without _sudo_ but it still fails. I tried _./release.sh_ but I get _permission denied_

![release](https://user-images.githubusercontent.com/51816834/64910191-8eafaa80-d70b-11e9-8d7e-bf506785c950.png)
With makefile for win64
![buildwin64](https://user-images.githubusercontent.com/51816834/64910048-e0a40080-d70a-11e9-8bd0-684f8b3c8621.png)
With makefile for win32
![buildwin32](https://user-images.githubusercontent.com/51816834/64910049-e13c9700-d70a-11e9-8f4e-8cbf7640b8d4.png)

**Comment by rom1v** [comment]
The "permission denied" is because you first call it with `sudo`. Remove all your build folders with `sudo` (`sudo rm -rf build_test`) then release without `sudo`.

**Comment by Erisaren** [comment]
I deleted the directories and tried release without  `sudo`.

![release2](https://user-images.githubusercontent.com/51816834/64917516-564eb180-d789-11e9-8f28-8d41a532cf52.png)

**Comment by rom1v** [comment]
Ok, this is a gradle issue now. Try switching to Java 8.

https://stackoverflow.com/a/53272420/1987178

_Please copy-paste the text instead of screenshot, so that errors can be indexed by search engines._

**Comment by Erisaren** [comment]
I edited the comments.
I think I fixed the issue with gradle. There was a _.gradle_ directory created as root and it was set on read only. I deleted the directory and launched `./release.sh` so it built a new directory. I also downgraded openjdk from 11 to 8.

I still get the same issue about SDL2.

> erisa@Kazuki:\~/scrcpy$ ./release.sh
> The Meson build system
> Version: 0.45.1
> Source dir: /home/erisa/scrcpy
> Build dir: /home/erisa/scrcpy/build_test
> Build type: native build
> Project name: scrcpy
> Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1\~18.04.1) 7.4.0")
> Build machine cpu family: x86_64
> Build machine cpu: x86_64
> Found pkg-config: /usr/bin/pkg-config (0.29.1)
> Native dependency libavformat found: YES 57.83.100
> Native dependency libavcodec found: YES 57.107.100
> Native dependency libavutil found: YES 55.78.100
> Native dependency sdl2 found: YES 2.0.10
> Configuring config.h using configuration
> Program ./scripts/build-wrapper.sh found: YES (/home/erisa/scrcpy/server/./scripts/build-wrapper.sh)
> server/meson.build:5: WARNING: Passed invalid keyword argument "console".
> WARNING: This will become a hard error in the future.
> WARNING: Unknown keyword arguments in target scrcpy-server: console
> Build targets in project: 8
> Found ninja-1.8.2 at /usr/bin/ninja
> ninja: Entering directory `build_test'
> [39/40] Generating scrcpy-server with a custom command.
> > Task :server:preBuild UP-TO-DATE
> > Task :server:preDebugBuild
> > Task :server:compileDebugAidl
> > Task :server:compileDebugRenderscript NO-SOURCE
> > Task :server:checkDebugManifest
> > Task :server:generateDebugBuildConfig
> > Task :server:prepareLintJar UP-TO-DATE
> > Task :server:generateDebugSources
> > Task :server:javaPreCompileDebug
> > Task :server:mainApkListPersistenceDebug
> > Task :server:generateDebugResValues
> > Task :server:generateDebugResources
> > Task :server:mergeDebugResources
> > Task :server:createDebugCompatibleScreenManifests
> > Task :server:processDebugManifest
> > Task :server:processDebugResources
> > Task :server:compileDebugJavaWithJavac
> > Task :server:compileDebugSources
> > Task :server:mergeDebugShaders
> > Task :server:compileDebugShaders
> > Task :server:generateDebugAssets
> > Task :server:mergeDebugAssets
> > Task :server:checkDebugDuplicateClasses
> > Task :server:mergeExtDexDebug
> > Task :server:mergeLibDexDebug
> > Task :server:transformClassesWithDexBuilderForDebug
> > Task :server:mergeProjectDexDebug
> > Task :server:validateSigningDebug
> > Task :server:signingConfigWriterDebug
> > Task :server:mergeDebugJniLibFolders
> > Task :server:transformNativeLibsWithMergeJniLibsForDebug
> > Task :server:transformNativeLibsWithStripDebugSymbolForDebug
> > Task :server:processDebugJavaRes NO-SOURCE
> > Task :server:transformResourcesWithMergeJavaResForDebug
> > Task :server:packageDebug
> > Task :server:assembleDebug
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD SUCCESSFUL in 3s
> 28 actionable tasks: 27 executed, 1 up-to-date
> [39/40] Running all tests.
> 1/5 test_cbuf                               OK       0.08 s
> 2/5 test_control_event_serialize            OK       0.09 s
> 3/5 test_device_event_deserialize           OK       0.10 s
> 4/5 test_queue                              OK       0.07 s
> 5/5 test_strutil                            OK       0.02 s
> 
> OK:         5
> FAIL:       0
> SKIP:       0
> TIMEOUT:    0
> 
> Full log written to /home/erisa/scrcpy/build_test/meson-logs/testlog.txt
> 
> > Task :server:lint
> Ran lint on variant release: 0 issues found
> Ran lint on variant debug: 0 issues found
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD SUCCESSFUL in 8s
> 33 actionable tasks: 8 executed, 25 up-to-date
> The Meson build system
> Version: 0.45.1
> Source dir: /home/erisa/scrcpy
> Build dir: /home/erisa/scrcpy/build_release
> Build type: native build
> Project name: scrcpy
> Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
> Build machine cpu family: x86_64
> Build machine cpu: x86_64
> Found pkg-config: /usr/bin/pkg-config (0.29.1)
> Native dependency libavformat found: YES 57.83.100
> Native dependency libavcodec found: YES 57.107.100
> Native dependency libavutil found: YES 55.78.100
> Native dependency sdl2 found: YES 2.0.10
> Configuring config.h using configuration
> Program ./scripts/build-wrapper.sh found: YES (/home/erisa/scrcpy/server/./scripts/build-wrapper.sh)
> server/meson.build:5: WARNING: Passed invalid keyword argument "console".
> WARNING: This will become a hard error in the future.
> WARNING: Unknown keyword arguments in target scrcpy-server: console
> Build targets in project: 8
> Found ninja-1.8.2 at /usr/bin/ninja
> [39/39] Generating scrcpy-server with a custom command.
> > Task :server:preBuild UP-TO-DATE
> > Task :server:preReleaseBuild UP-TO-DATE
> > Task :server:compileReleaseAidl UP-TO-DATE
> > Task :server:compileReleaseRenderscript NO-SOURCE
> > Task :server:checkReleaseManifest UP-TO-DATE
> > Task :server:generateReleaseBuildConfig UP-TO-DATE
> > Task :server:prepareLintJar UP-TO-DATE
> > Task :server:generateReleaseSources UP-TO-DATE
> > Task :server:javaPreCompileRelease UP-TO-DATE
> > Task :server:mainApkListPersistenceRelease UP-TO-DATE
> > Task :server:generateReleaseResValues UP-TO-DATE
> > Task :server:generateReleaseResources UP-TO-DATE
> > Task :server:mergeReleaseResources UP-TO-DATE
> > Task :server:createReleaseCompatibleScreenManifests UP-TO-DATE
> > Task :server:processReleaseManifest UP-TO-DATE
> > Task :server:processReleaseResources UP-TO-DATE
> > Task :server:compileReleaseJavaWithJavac UP-TO-DATE
> > Task :server:compileReleaseSources UP-TO-DATE
> > Task :server:lintVitalRelease
> > Task :server:checkReleaseDuplicateClasses UP-TO-DATE
> > Task :server:mergeExtDexRelease UP-TO-DATE
> > Task :server:transformClassesWithDexBuilderForRelease UP-TO-DATE
> > Task :server:mergeDexRelease UP-TO-DATE
> > Task :server:mergeReleaseShaders UP-TO-DATE
> > Task :server:compileReleaseShaders UP-TO-DATE
> > Task :server:generateReleaseAssets UP-TO-DATE
> > Task :server:mergeReleaseAssets UP-TO-DATE
> > Task :server:signingConfigWriterRelease UP-TO-DATE
> > Task :server:mergeReleaseJniLibFolders UP-TO-DATE
> > Task :server:transformNativeLibsWithMergeJniLibsForRelease UP-TO-DATE
> > Task :server:transformNativeLibsWithStripDebugSymbolForRelease UP-TO-DATE
> > Task :server:processReleaseJavaRes NO-SOURCE
> > Task :server:transformResourcesWithMergeJavaResForRelease UP-TO-DATE
> > Task :server:packageRelease UP-TO-DATE
> > Task :server:assembleRelease
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD SUCCESSFUL in 2s
> 27 actionable tasks: 1 executed, 26 up-to-date
> /home/erisa/scrcpy
> ./gradlew clean
> 
> BUILD SUCCESSFUL in 878ms
> 2 actionable tasks: 2 executed
> rm -rf "build-server" "build-win32" "build-win64" \
>            "build-win32-noconsole" "build-win64-noconsole" "dist"
> [ -d "build-server" ] || ( mkdir "build-server" && \
> 	meson "build-server" \
> 		--buildtype release -Dcompile_app=false )
> The Meson build system
> Version: 0.45.1
> Source dir: /home/erisa/scrcpy
> Build dir: /home/erisa/scrcpy/build-server
> Build type: native build
> Project name: scrcpy
> Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
> Build machine cpu family: x86_64
> Build machine cpu: x86_64
> Program ./scripts/build-wrapper.sh found: YES (/home/erisa/scrcpy/server/./scripts/build-wrapper.sh)
> server/meson.build:5: WARNING: Passed invalid keyword argument "console".
> WARNING: This will become a hard error in the future.
> WARNING: Unknown keyword arguments in target scrcpy-server: console
> Build targets in project: 2
> Found ninja-1.8.2 at /usr/bin/ninja
> ninja -C "build-server"
> ninja: Entering directory `build-server'
> [1/1] Generating scrcpy-server with a custom command.
> > Task :server:preBuild UP-TO-DATE
> > Task :server:preReleaseBuild
> > Task :server:compileReleaseAidl
> > Task :server:compileReleaseRenderscript NO-SOURCE
> > Task :server:checkReleaseManifest
> > Task :server:generateReleaseBuildConfig
> > Task :server:prepareLintJar UP-TO-DATE
> > Task :server:generateReleaseSources
> > Task :server:javaPreCompileRelease
> > Task :server:mainApkListPersistenceRelease
> > Task :server:generateReleaseResValues
> > Task :server:generateReleaseResources
> > Task :server:mergeReleaseResources
> > Task :server:createReleaseCompatibleScreenManifests
> > Task :server:processReleaseManifest
> > Task :server:processReleaseResources
> > Task :server:compileReleaseJavaWithJavac
> > Task :server:compileReleaseSources
> > Task :server:lintVitalRelease
> > Task :server:checkReleaseDuplicateClasses
> > Task :server:mergeExtDexRelease
> > Task :server:transformClassesWithDexBuilderForRelease
> > Task :server:mergeDexRelease
> > Task :server:mergeReleaseShaders
> > Task :server:compileReleaseShaders
> > Task :server:generateReleaseAssets
> > Task :server:mergeReleaseAssets
> > Task :server:signingConfigWriterRelease
> > Task :server:mergeReleaseJniLibFolders
> > Task :server:transformNativeLibsWithMergeJniLibsForRelease
> > Task :server:transformNativeLibsWithStripDebugSymbolForRelease
> > Task :server:processReleaseJavaRes NO-SOURCE
> > Task :server:transformResourcesWithMergeJavaResForRelease
> > Task :server:packageRelease
> > Task :server:assembleRelease
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/5.5.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD SUCCESSFUL in 4s
> 27 actionable tasks: 26 executed, 1 up-to-date
> make -C prebuilt-deps prepare-win32
> make[1]: Entering directory '/home/erisa/scrcpy/prebuilt-deps'
> SDL2-2.0.10: found
> ffmpeg-4.1.4-win32-shared: found
> ffmpeg-4.1.4-win32-dev: found
> platform-tools: found
> make[1]: Leaving directory '/home/erisa/scrcpy/prebuilt-deps'
> [ -d "build-win32" ] || ( mkdir "build-win32" && \
> 	meson "build-win32" \
> 		--cross-file cross_win32.txt \
> 		--buildtype release --strip -Db_lto=true \
> 		-Dcrossbuild_windows=true \
> 		-Dcompile_server=false \
> 		-Dportable=true )
> The Meson build system
> Version: 0.45.1
> Source dir: /home/erisa/scrcpy
> Build dir: /home/erisa/scrcpy/build-win32
> Build type: cross build
> Project name: scrcpy
> Native C compiler: cc (gcc 7.4.0 "cc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0")
> Cross C compiler: /usr/bin/i686-w64-mingw32-gcc (gcc 7.3.0)
> Host machine cpu family: x86
> Host machine cpu: i686
> Target machine cpu family: x86
> Target machine cpu: i686
> Build machine cpu family: x86_64
> Build machine cpu: x86_64
> 
> app/meson.build:45:4: ERROR: C library 'SDL2' not found
> 
> A full log can be found at /home/erisa/scrcpy/build-win32/meson-logs/meson-log.txt
> Makefile.CrossWindows:54: recipe for target 'build-win32' failed
> make: *** [build-win32] Error 1
>

**Comment by rom1v** [comment]
```
sha256sum prebuilt-deps/SDL2-devel-2.0.10-mingw.tar.gz
find prebuilt-deps/SDL2-2.0.10/
```
?

**Comment by Erisaren** [comment]
I deleted the sdl2 dev zip file at first because there was a checksum mismatch and it was a zero byte file, then I added the sdl2 file I downloaded from libsdl website and well, it seems since it was not the dev version, it caused some issues, then the gradle issue and running with sudo worsened it.

I deleted the sdl2 folder and upon launching `./release.sh` it downloaded the sdl2 dev zip file again and the checksum matched, finally.

Sorry for all the bother and thank you for your patience.
